packages/js/src/ui/components/elements/Preferences/ScheduleTable.tsx

```diff
@@ -0,0 +1,239 @@
+import { TimeSelect } from '../../primitives';
+import { Switch } from '../../primitives/Switch';
+import { DayScheduleCopy } from './DayScheduleCopy';
+import { weekDays } from './utils';
+
+type ScheduleTableHeaderProps = {
+  schedule?: Schedule;
+  children: JSX.Element;
+};
+
+const ScheduleTableHeader = (props: ScheduleTableHeaderProps) => {
+  const style = useStyle();
+  return (
+    <div
+      class={style({
+        key: 'scheduleTableHeader',
+        className: 'nt-flex nt-gap-3',
+        context: { schedule: props.schedule } satisfies Parameters<AppearanceCallback['scheduleTableHeader']>[0],
+      })}
+    >
+      {props.children}
+    </div>
+  );
+};
+
+type ScheduleTableHeaderColumnProps = {
+  schedule?: Schedule;
+  children: JSX.Element;
+  class?: string;
+  dataLocalization?: string;
+};
+
+const ScheduleTableHeaderColumn = (props: ScheduleTableHeaderColumnProps) => {
+  const style = useStyle();
+  return (
+    <div
+      class={style({
+        key: 'scheduleHeaderColumn',
+        className: cn('nt-text-sm nt-text-neutral-600 nt-truncate nt-text-start', props.class),
+        context: { schedule: props.schedule } satisfies Parameters<AppearanceCallback['scheduleHeaderColumn']>[0],
+      })}
+      data-localization={props.dataLocalization}
+    >
+      {props.children}
+    </div>
+  );
+};
+
+type ScheduleTableBodyProps = {
+  schedule?: Schedule;
+  children: JSX.Element;
+};
+
+const ScheduleTableBody = (props: ScheduleTableBodyProps) => {
+  const style = useStyle();
+  return (
+    <div
+      class={style({
+        key: 'scheduleTableBody',
+        className: 'nt-flex nt-flex-col nt-gap-1',
+        context: { schedule: props.schedule } satisfies Parameters<AppearanceCallback['scheduleTableBody']>[0],
+      })}
+    >
+      {props.children}
+    </div>
+  );
+};
+
+type ScheduleTableRowProps = {
+  schedule?: Schedule;
+  children: JSX.Element;
+};
+
+const ScheduleTableRow = (props: ScheduleTableRowProps) => {
+  const style = useStyle();
+  return (
+    <div
+      class={style({
+        key: 'scheduleBodyRow',
+        className: 'nt-flex nt-gap-3',
+        context: { schedule: props.schedule } satisfies Parameters<AppearanceCallback['scheduleBodyRow']>[0],
+      })}
+    >
+      {props.children}
+    </div>
+  );
+};
+
+type ScheduleTableCellProps = {
+  schedule?: Schedule;
+  children: JSX.Element;
+  class?: string;
+};
+const ScheduleBodyColumn = (props: ScheduleTableCellProps) => {
+  const style = useStyle();
+  return (
+    <div
+      class={style({
+        key: 'scheduleBodyColumn',
+        className: cn('nt-text-sm nt-text-neutral-600', props.class),
+        context: { schedule: props.schedule } satisfies Parameters<AppearanceCallback['scheduleBodyColumn']>[0],
+      })}
+    >
+      {props.children}
+    </div>
+  );
+};
+
+type ScheduleTableProps = {
+  globalPreference?: Preference;
+};
+
+export const ScheduleTable = (props: ScheduleTableProps) => {
```

Comment:
the table is a list of all week days, each includes the toggle, copy, time start/end components